plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/standard/StandardDeployCommandHandler.java

```diff
@@ -40,6 +44,32 @@ public StandardDeployCommandHandler() {
     super(AnalyticsEvents.APP_ENGINE_DEPLOY_STANDARD);
   }
 
+  @Override
+  protected boolean checkProject(Shell shell, IProject project) throws CoreException {
+    if (WebProjectUtil.hasJsps(project)) {
+      IJavaProject javaProject = JavaCore.create(project);
+      IVMInstall vmInstall = JavaRuntime.getVMInstall(javaProject);
+      if (!JreDetector.isDevelopmentKit(vmInstall)) {
+        return MessageDialog.openQuestion(
```

Comment:
anyway to change Yes/No to Deploy/Cancel?

Reply:
Will do.